### ISSUE TITLE
Mob Masher Upgrades disappearing on Server Restart

### ISSUE THREAD

**Issue by LarkinOmega** [Sunekaer/stoneBlock#534]
## Bug Report
Mob Masher Upgrades placed in Mob Masher disappear upon server restart in Single Player.

## Steps to Reproduce (for bugs)
1. Place Upgrade in Mob Masher
2. Restart Server
3. Observe that Upgrades are no longer in the Mob Masher

## Client Information
* Modpack Version: 1.0.26
* Server/LAN/Single Player: Single Player
* Optifine Installed: No
* Shaders Enabled: None

**Comment by Sunekaer** [comment]
just tested, all upgrades are still in my mob masher after a restart.
Could you try and test again?

**Comment by LarkinOmega** [comment]
Yes, it's still doing it for me. I have 3 mashers placed next to each other, so it's very apparent that they've gone missing from all of them.

Not sure how to proceed from here. Let me know if I can provide you with anything else.

**Comment by LarkinOmega** [comment]
Found the problem. When Ender IO Redstone conduits are forcibly attached, the upgrades disappear on server restart.

Interaction bug between Mob Mashers and Ender IO.

**Comment by LarkinOmega** [comment]
> Found the problem. When Ender IO Redstone conduits are forcibly attached, the upgrades disappear on server restart.

This is incorrect. I removed the direct connection and instead sent a redstone signal through a solid block. Upon login today, the upgrades were once again missing.

I applied upgrades and tested the following, reapplying upgrades as needed:

1) Turned OFF the Redstone Signal, logged out of the world
> Upgrades retained

2) Turned OFF the Redstone Signal, logged out of the world, and closed the client
> Upgrades retained

3) Turned ON the Redstone Signal, logged out of the world
> **Upgrades Removed**

4) Removed Ender I/O cables, used vanilla Redstone applied directly, Turned On, logged out of the world
> Upgrades retained

5) Vanilla Redstone, Indirectly applied, Turned On, Logged out of the World
> Upgrades retained

6) Used Ender I/O Redstone cables buffered by Vanilla Redstone, Turned On, Logged out of the world
> Upgrades retained

7) Switch to RF Tools Redstone direct connection, Turned on, Logged out of the world
> Upgrades retained

There appears to be only one failure case. A redstone signal generated from Ender I/O cables, applied to the Mob Masher, directly or indirectly, upon server restart, causes Mob Masher Upgrades to disappear.

I'm not sure this bug deserved the time I spent isolating it TBH. It is very niche, and just having knowledge of it makes it's existence rather moot and easily worked around.
